File: kotlin/src/main/kotlin/com/github/princesslana/smalld/examples/PingBot.kt
package com.github.princesslana.smalld.examples

val pingBot = bot {
  fun onMessageCreate(msg: Message) {
    when (msg.content) {
      "++ping" -> post("/channels/${msg.channelId}/messages", CreateMessage("pong"))
    }
  }

  onGatewayPayload { p ->
    when (p.t) {
      "MESSAGE_CREATE" -> onMessageCreate(gson.fromJson(p.d, Message::class.java))
    }
  }
}

fun main(args: Array<String>) = pingBot.run(System.getenv("SMALLD_TOKEN"))

File: ruby/lib/pingbot.rb
require_relative 'bot'

$ping_bot = Bot.new do
  on_message_create do |msg|
    post "/channels/#{msg.channel_id}/messages", content: 'pong' if msg.content == '++ping'
  end
end

$ping_bot.run token: ENV['SMALLD_TOKEN'] if __FILE__ == $0

